ghastly/ghastly.py

```diff
@@ -0,0 +1,292 @@
+import numpy as np
+import openmc
+import ghastly
+from jinja2 import Environment, PackageLoader
+from ghastly import read_input
+from ghastly import region
+from lammps import lammps
+
+env = Environment(loader=PackageLoader('ghastly','templates'))
+
+def fill_core(input_file, rough_pf):
+    '''
+    This function uses OpenMC's model.pack_spheres function to "rough" pack
+    the full core region defined by the Ghastly input file, then uses a series
+    of Jinja templates to create a LAMMPS input file that will run a physics
+    simulation, modeling not only the settling of the roughly packed pebbles
+    due to gravtiy, but also handle the addtion of any extra pebbles needed to
+    reach the target number of pebbles needed.
+
+    Parameters
+    ----------
+    input_file : str
+        Name of ghastly input file, which should be in JSON format, following
+        the layout and parameter naming convention given in the example inputs.
+    rough_pf : float
+        Packing fraction used in OpenMC packing function.  Can match the target
+        packing fraction provided in the input file, but users should take care 
```

Comment:
Both you and Luke asked this question, so I definitely need to make this clearer - but yes, if you add a much higher rough_pf for doing OpenMC packing than you give for the target pf (like, if you only wanted 30% of the core to have pebbles, for some reason, but you supply a rough_pf of 60%) you could "overfill" the core.

In the future this will change a bit, because I'd like to adjust it to auto-fill the "discharge" region to 60% no matter what.  Then, the packing fractions (both the rough pf and the 'target' pf in the input file) would only be referring the the main body of the core (however, you can still have this "overfilling" issue).  It's also possible I remove rough_pf later, and have it always match the "target", but for now while I'm developing things I think it's potentially useful to have them separate.